src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurDetailLineGroupForCollectorHelper.java

```diff
@@ -79,4 +97,18 @@ public String formatDate(Date value, String format) {
         return dateTimeService.toString(value, format);
     }
 
+    public ConcurRequestedCashAdvance getRequestedCashAdvanceByCashAdvanceKey(String cashAdvanceKey) {
+        Collection<ConcurRequestedCashAdvance> requestedCashAdvances = concurRequestedCashAdvanceService
+                .findConcurRequestedCashAdvanceByCashAdvanceKey(cashAdvanceKey);
+        if (CollectionUtils.size(requestedCashAdvances) == 1) {

```

Comment:
There should be no more than one ConcurRequestedCashAdvance object per cash advance key. The current implementation of this ConcurRequestedCashAdvanceService method is just returning the results of a BusinessObjectService.findMatching() operation, which returns a Collection.

Perhaps ConcurRequestedCashAdvanceService could be updated to return zero or one result instead, but that is beyond the scope of this PR and should be deferred to a future enhancement.